### PR TITLE
Client status/buttons wording changes

### PR DESCRIPTION
**Pull request by loneil**
# Description
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-926

Change to a 3-state client status (see ticket). Alterations to My Applications, and API Access step 1.

Refactor status to component instead of the util function.
Wanted to do a functional single file component (https://itnext.io/whats-the-deal-with-functional-components-in-vue-js-513a31eb72b0), but requires computed field calculations based on passed in props so not appropriate for this use.

Update vuetify

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-167

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-167

File: app/frontend/src/components/ApplicationList.vue [hunk 1]
```diff
@@ -45,23 +49,26 @@
...
+import ClientStatus from '@/components/apiAccess/ClientStatus.vue';
 
 export default {
   name: 'ApplicationList',
+  components: {
+    ClientStatus
+  },
+  data() {
+    return {
+      envs: KcEnv
```
Comment (jujaga) [inline app/frontend/src/components/ApplicationList.vue:61]: This should be a computed - especially if it's supposed to be a constant resource.

File: app/frontend/src/components/apiAccess/ClientStatus.vue [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<template>
+  <div>
+    {{ env }}:
```
Comment (jujaga) [inline app/frontend/src/components/apiAccess/ClientStatus.vue:3]: Should this be in its own span object?

File: app/frontend/src/components/apiAccess/ClientStatus.vue [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    <span :class="display.class">{{display.text}}</span>
```
Comment (jujaga) [inline app/frontend/src/components/apiAccess/ClientStatus.vue:4]: Space between the curly braces and content

File: app/frontend/src/components/apiAccess/ClientStatus.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          } else {
+            label = this.clientStatuses.dev ? KcClientStatus.READY : KcClientStatus.NOT;
+          }
+        } else if(this.env === KcEnv.PROD) {
+          if(this.clientStatuses.prod) {
+            label = KcClientStatus.CREATED;
+          } else {
+            label = this.clientStatuses.test ? KcClientStatus.READY : KcClientStatus.NOT;
+          }
+        }
```
Comment (jujaga) [inline app/frontend/src/components/apiAccess/ClientStatus.vue:44]: If and ( should have a space between.

Half tempted to suggest that this should be a switch instead, considering there's a few repeated else statements. Another option is to just have an auxiliary function which takes in the environment and status as parameters, and outputs the expected constant label instead.

**Comment by loneil** [comment]
Feel free to delete clients for WORG and TEMP (careful only mess with those) in YAMS realms to test out statuses

File: app/frontend/src/components/apiAccess/Step1.vue [hunk 5]
```diff
@@ -70,23 +79,25 @@
...
 export default {
   name: 'ApiAccessStep1',
+  components: {
+    ClientStatus
+  },
   data() {
     return {
       KcClientStatus: KcClientStatus,
-      env: KcEnv
+      envs: KcEnv
```
Comment (jujaga) [inline app/frontend/src/components/apiAccess/Step1.vue:93]: These should probably be computed as they are meant to be constants.

File: app/frontend/src/components/ApplicationList.vue (hunk 1 above)
Comment (loneil) [reply app/frontend/src/components/ApplicationList.vue:61]: I guess

File: app/frontend/src/components/apiAccess/ClientStatus.vue (hunk 2 above)
Comment (loneil) [reply app/frontend/src/components/apiAccess/ClientStatus.vue:3]: No doesn't need to be inline or block level, just text

File: app/frontend/src/components/apiAccess/ClientStatus.vue (hunk 3 above)
Comment (loneil) [reply app/frontend/src/components/apiAccess/ClientStatus.vue:4]: Fixed

File: app/frontend/src/components/apiAccess/ClientStatus.vue (hunk 4 above)
Comment (loneil) [reply app/frontend/src/components/apiAccess/ClientStatus.vue:44]: Fixed spacing, would do switch if it was 1 more else if. Leaving as ifs

File: app/frontend/src/components/apiAccess/Step1.vue (hunk 5 above)
Comment (loneil) [reply app/frontend/src/components/apiAccess/Step1.vue:93]: Fixed

**Pull request merged**